lib/nerves_network/dhcp_manager.ex

```diff
@@ -277,19 +266,47 @@ defmodule Nerves.Network.DHCPManager do
     state
   end
 
-  @spec configure(t, Types.udhcp_info) :: t
+  defp setup_iface(state, info) do
+    case Nerves.NetworkInterface.setup(state.ifname, info) do
+      :ok -> :ok
+      {:error, :eexist} -> :ok
+        #It may very often happen that at the renew time we would receive the lease of the very same IP address...
+        #In such a case whilst adding already existent IP address to the network interface we shall receive 'error exists'.
+        #It definitely is non-critical situation and actually confirms that we do not have to take any action.
+    end
+  end
+
+  defp remove_old_ip(state, info) do
+    old_ip = info[:"old_ipv4_address"] || ""
+    new_ip = info[:"ipv4_address"] || ""
```

Comment:
~~~elixir
defp remove_old_ip(%{old_ipv4_address: old_ip, ipv4_address: old_ip}), do: :ok
defp remove_old_ip(%{old_ipv4_address: nil}), do: :ok
defp remove_old_ip(%{old_ipv4_address: old_ip, ifname: ifname}) do
  Logger.debug fn -> "Removing ipv4 address = #{inspect old_ip} from #{inspect ifname}" end
  Nerves.NetworkInterface.setup(ifname, %{:"-ipv4_address" => old_ip})
end
~~~

This might be an option too.